src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -359,12 +365,17 @@
       "metrics" {:type :json
                  :queryable? false
                  :field {:select [(h/row-to-json :t)]
-                         :from [[{:select [[:metrics :data] [(hsql-hash-as-str :hash) :href]]} :t]]}
+                         :from [[{:select
+                                  [[(h/coalesce :metrics
+                                                (h/scast :metrics_json (jsonb-type))) :data]
```

Comment:
still testing this with a large dataset. my remaining concern is that this cast could be very expensive for queries returning large numbers of reports.